src/python/m5/ext/pystats/jsonloader.py

```diff
@@ -0,0 +1,55 @@
+from .simstat import SimStat
+from .statistic import Scalar, Distribution, Accumulator
+from .group import Group, Vector
+import json
+
+def json_loader(path: str) -> SimStat:
```

Comment:
maybe just call this `load`? Is there a way it could have the *same* interface as `json.load`?

Reply:
Maybe we don't even need this wrapper function then. The user can just call 
> simstat_obj = json.load(f, object_hook=__json_to_simstat)
like in line 9.

Reply:
I like the idea of hiding the `object_hook` from "regular" users.

Reply:
Ok. I will look into subclassing the JSONDecoder then.

Reply:
Would something like json.loads(f, cls=JSONLoader) be acceptable, where JSONLoader is a subclass of JSONDecoder with object_hooks overwritten? I am trying to think of a way that we could hide the underlying "JSONLoader" stuff from user entirely